### ISSUE TITLE
Support multiple languages

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [RagnarGrootKoerkamp/BAPCtools#109]
Maybe something like this:
Have a `language: <en|nl|...>` in `contest.yaml` and/or `problem.yaml`, so that `bt` will look for `problem.<language>.tex` instead of the default `problem.en.tex`.

A clean way to change the latex constants would be to have language specific `en.tex`, `nl.tex`, ... includes that contain all the hardcoded texts as commands.

Most of the [diff to the NIO fork](https://github.com/RagnarGrootKoerkamp/BAPCtools/compare/master...ludopulles:nio) should be resolved by this.

CC @ludopulles

**Comment by mzuenni** [comment]
maybe use [translator](https://ctan.org/pkg/translator) for latex with `<language>.dict` files.

**Comment by thorehusfeldt** [comment]
Regarding specification, `language` in `problem.yaml` would be very close to the already-existing `languages` (which specifies programming languages.)

Suggestion: according to the specification, the `name` key in `problem.yaml` already takes map; it can look like this:
```yaml
name:
  en: Danish Pastry
  nl: Stroopwafel
  de: Kopenhagener Gebäck
  da: Wienerbrød
```
From the keys in that map, we can infer the four languages and expect `problem.nl.tex` to exist (and complain otherwise).

**Comment by RagnarGrootKoerkamp** [comment]
I'd be happy to review a PR for this. I don't imagine this would be too hard to support once you know where to look. I could probably give some pointers if you'd like.

(We don't use/need this for BAPC, and sadly I don't really have time and motivation to write new code.)

**Comment by thorehusfeldt** [comment]
OK, I like this idea. My current ambition is to use BAPCtools for the next Will Code for Drinks, scheduled for November 2023; but I need to find my way around it a bit more before I commit to anything. 

I forked the repo for now and implemented the default grader, which doesn’t really have much to do with statement languages to be honest.

**Comment by thorehusfeldt** [comment]
First attempt and discussion at https://github.com/RagnarGrootKoerkamp/BAPCtools/discussions/254

**Comment by thorehusfeldt** [comment]
@RagnarGrootKoerkamp : my https://github.com/thorehusfeldt/BAPCtools/tree/i18n branch now supports some basic functionality, such as 

```
myproblem> bt pdf
myproblem> bt pdf --language nl
cd ..
mycontest> bt pdf
mycontest> bt pdf --language nl
```
which create `myproblem.XY.pdf` and `mycontest.XY.pdf` as expected.

The `problem.yaml` file now prefers the map
```yaml
  name:
    en: Detour
```
to the string
```yaml
name: Detour
```
`problem.py` still understands “stringy” version (and translates it internally in `problem.settings`, but both the skeleton and `new_problem` wake up with the “mappy” version.

`bt new_problem` now begins by asking for problem statement languages; this could of course interact with a specification file such as `locales.yaml` on contest or user level. `bt rename_problem` iterates over all languages when renaming.

Tell me if you think this goes in the right direction. What would be useful now would be a recent BAPC source directory so that I can check backwards compatibility.

## In progress

I haven’t thought solutions through, ideally I at least shouldn’t break  anything.
Currently, the command
```
bt solutions
```
works as before from a problemdir, except for producing `solution.en.pdf`. (`bt solution` from a contestdir is currently broken.)

**Comment by thorehusfeldt** [comment]
Thanks for your quick reply, your enthusiasm, and the quality of your response. 

> The warning for language in problem.yaml that's missing from problem.xy.tex should also be an error I'd say.

The specification is explicit only about one of them. But I’m happy to make both errors, just for symmetry. 

But I’ll play around with this branch a bit more before I send a PR. No rush.

**Comment by mpsijm** [comment]
Implemented in #256 :tada:
